lib/std/secrets.nim

```diff
@@ -0,0 +1,98 @@
+import os
+
+runnableExamples:
+  doAssert urandom(0).len == 0
+  doAssert urandom(10).len == 10
+  doAssert urandom(20).len == 20
+  doAssert urandom(120).len == 120
+
+
+when defined(posix):
+  import posix
+
+  template processReadBytes(readBytes: int, p: pointer) =
+    if readBytes == 0:
+      break
+    elif readBytes > 0:
+      inc(result, readBytes)
+      cast[ptr pointer](p)[] = cast[pointer](cast[ByteAddress](p) + readBytes)
+    else:
+      if osLastError().int in {EINTR, EAGAIN}:
+        discard
+      else:
+        result = -1
+        break
+
+  proc getDevUrandom(p: pointer, size: int): int =
+    let fd = posix.open("/dev/urandom", O_RDONLY)
```

Comment:
If avaiable, use an OS API instead. See https://insanecoding.blogspot.com/2014/05/a-good-idea-with-bad-usage-devurandom.html for more information.

Reply:
Yeah, It's convenient for me to test specific system call in Windows and Linux VM. But I can't test a function specific to openbsd/freebsd and OSX. So I haven't work on them because I don't want to use CI to debug.

Reply:
@xflywind there's https://www.macincloud.com/ for OSX (I've used it occasionally even though i have osx, when i needed to test other OSX versions); i can help also if needed for osx

for openbsd/freebsd, both should also have urandom

Reply:
there are nimble packages that provide these procs. See nim-crypto

Reply:
https://github.com/cheatfate/nimcrypto/blob/master/nimcrypto/sysrand.nim